src/Axiom/Set/Properties.agda

```diff
@@ -64,9 +68,28 @@ module _ {l : List A} {a} where
   ∈-fromList⁻ : a ∈ fromList l → a ∈ˡ l
   ∈-fromList⁻ = from ∈-fromList
 
-  ∈-fromList⁺ : ∀ {l : List A} {a} → a ∈ˡ l → a ∈ fromList l
+  ∈-fromList⁺ : a ∈ˡ l → a ∈ fromList l
   ∈-fromList⁺ = to ∈-fromList
 
+module _ {f : A → B} {l : List A} {b : B} where
+  ∈-fromList-map⁺ : b ∈ fromList (Data.List.map f l) → ∃[ a ] a ∈ˡ l × b ≡ f a
+  ∈-fromList-map⁺ = (∈-map⁻ f) ∘ ∈-fromList⁻
+
+  ∈-fromList-map⁻ : ∃[ x ] x ∈ˡ l × b ≡ f x → b ∈ fromList (Data.List.map f l)
+  ∈-fromList-map⁻ (x , x∈ , b≡) = ∈-fromList⁺ (∈-map⁺ (x , b≡ , x∈))
```

Comment:
Okay, I'll remove them again... I reproved them when resuming work on this PR because they were useful when proving things that required keeping a lot of logic in my head at once.  Without these abstractions, the slightly higher complexity somehow made it significantly more difficult to get the proof right.